sdks/smart-wallet-sdk/src/smartWallet.ts

```diff
@@ -1,40 +1,60 @@
 import { ChainId } from '@uniswap/sdk-core'
-import { encodeAbiParameters, encodeFunctionData } from 'viem'
+import { encodeFunctionData } from 'viem'
 
-import { abi } from '../abis/MinimalDelegation.json'
+import { abi } from '../abis/MinimalDelegationEntry.json'
 
-import { MODE_TYPE_ABI_PARAMETERS, ModeType, SMART_WALLET_ADDRESSES } from './constants'
-import { Call, MethodParameters, ExecuteOptions, AdvancedCall } from './types'
+import { ModeType, SMART_WALLET_ADDRESSES } from './constants'
+import { Call, MethodParameters, ExecuteOptions } from './types'
 import { CallPlanner } from './utils'
+import { BatchedCallPlanner } from './utils/batchedCallPlanner'
 
 /**
- * Main SDK class for interacting with ERC7821-compatible smart wallets
+ * Main SDK class for interacting with Uniswap smart wallet contracts
  */
 export class SmartWallet {
   /**
    * Creates method parameters for executing a simple batch of calls through a smart wallet
-   * @dev does not support opData
    * @param calls Array of calls to encode
+   * @param options Basic options for the execution
    * @returns Method parameters with calldata and value
    */
-  public static encodeCalls(calls: Call[], options: ExecuteOptions = {}): MethodParameters {
+  public static encodeBatchedCall(calls: Call[], options: ExecuteOptions = {}): MethodParameters {
     const mode = this.getModeFromOptions(options)
     if(mode != ModeType.BATCHED_CALL && mode != ModeType.BATCHED_CALL_CAN_REVERT) {
       throw new Error(`Invalid mode: ${mode}`)
     }
```

Comment:
So yeah I think this entrypoint should just be the direct encoding, not 7821 encoding

Reply:
so it seems like you can just remove these lines/not support a mode being passed in